front/lib/api/assistant/messages.ts

```diff
@@ -127,6 +128,23 @@ export async function batchRenderAgentMessages(
 ): Promise<{ m: AgentMessageType; rank: number; version: number }[]> {
   const agentMessages = messages.filter((m) => !!m.agentMessage);
 
+  const retrievalActionByAgentMessageId = (
```

Comment:
This query should be part of the Promise.all below.

We had a Promise.all with one query per action type. If we keep it here we create a sequence of one query for retrieval and then a Promise.all with all actions + among this Promise.all one function which is not even async.

Let's have the Promise.all run this query and return the `retrievalActionByAgentMessageId` + move the sync generation of `agentRetrievalActions` outside of the Promise.all :pray: